pkg/collector/corechecks/ebpf/probe/ebpfcheck/probe_test.go

```diff
@@ -306,13 +314,6 @@ func TestHashMapNumberOfEntriesNoExtraAllocations(t *testing.T) {
 					require.LessOrEqual(t, allocs, 8.0) // Multiple batches mean we need to use a map to keep track of the keys, that causes allocations for the values
 				})
 			}
-
-			t.Run("MainFunction", func(t *testing.T) {
```

Comment:
I removed this because the helper based solution is not alloc-free. So the main function is not really alloc free anymore